src/Application/Common/Interfaces/ICoreApiService.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Codidact.Authentication.Application.Common.Interfaces
+{
+    /// <summary>
+    /// Service that connect to the Core Codidact API
+    /// </summary>
+    public interface ICoreApiService
+    {
+        Task<bool> CreateMember(string displayName, long userId);
```

Comment:
> The core application asks for the information it needs, this might be too complicated though.

You mean after a handshake the Core API asks for whatever data it needs? It can be done.

> We create some sort of transfer object which has a DisplayName field. This is not too much additional code but we might have to go with the first option at some point.

Via the token? But the problem is I don't have any conformation that the member has been successfully created. What if we decide the display name is unique or something or userid already has a member due to an error.

> We just leave it as it is now and change it later when we know more precisely what we need.

I wrote down a gRPC API actually for this thing on core and tried to see how to work with it. ( You can take a look https://github.com/misha130/core/tree/api/src/Api ) 
Its a lot of work. You need to have a repository for the protos I'd really just leave it as is without the confirmation for this PR. Just let this PR be about the register page, not the API.